### ISSUE TITLE
[java] StringToString can not detect the case: getStringMethod().toString()

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2530]
**Affects PMD Version:** 
6.22.0

**Rule:**
[StringToString](https://pmd.github.io/latest/pmd_rules_java_performance.html#stringtostring)

**Description:**
StringToString can not detect the case: <code>getStringMethod().toString()</code>
This rule is implemented in File: pmd-java-6.22.0-sources.jar!\net\sourceforge\pmd\lang\java\rule\performance\StringToStringRule.java
Firstly, this method visits ASTVariableDeclaratorId node and gets variables of type <code>String</code> or <code>String[]</code>, and then check whether these variables call <code>toString()</code> method.
Calling <code>toString()</code> by another method that return <code>String</code> may be ignored.

**Code Sample demonstrating the issue:**
```java
public class Test{
	public String foo(){
		String bar = "foobar";
		return bar;
	}
	
	public void test(){
		String fooStr = foo().toString();
	}
}
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI

**Comment by adangel** [comment]
Note: The fix in #2659 will only detect such methods that return String *within the same file*. That means, this implementation doesn't detect cases when toString() call is chained to a method returning String which is not declared in  the class having the call.

That means, the given example in the issue is fixed, but there might still be other (more complex) false negatives, which PMD doesn't detect.
